### PR TITLE
Feature/grpc

### PR DESCRIPTION
**Pull request by ajax-makarytskyi-o**
gRPC controller was added.

### CODE REVIEW

File: domainservice/src/main/kotlin/com/makarytskyi/rentcar/kafka/CreateOrderKafkaProducer.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.springframework.stereotype.Component
+import reactor.core.publisher.Mono
+import reactor.kafka.sender.KafkaSender
+import reactor.kafka.sender.SenderRecord
+import reactor.kotlin.core.publisher.toMono
+
+@Component
+class CreateOrderKafkaProducer(
+    private val createOrderKafkaSender: KafkaSender<String, ByteArray>
```
Comment (ajax-marchenko-s) [inline domainservice/src/main/kotlin/com/makarytskyi/rentcar/kafka/CreateOrderKafkaProducer.kt]: add comma

File: gateway/src/main/kotlin/com/makarytskyi/gateway/config/NatsClient.kt [hunk 2]
```diff
@@ -2,17 +2,23 @@ package com.makarytskyi.gateway.config
...
+import reactor.core.publisher.Flux
 import reactor.core.publisher.Mono
 
 @Component
 class NatsClient(
     private val natsConnection: Connection,
     @Value("\${nats.timeout}")
-    private val timeoutDuration: Long
+    private val timeoutDuration: Long,
+    private val dispatcher: Dispatcher
```
Comment (ajax-marchenko-s) [inline gateway/src/main/kotlin/com/makarytskyi/gateway/config/NatsClient.kt]: add comma

File: domainservice/src/main/kotlin/com/makarytskyi/rentcar/service/impl/OrderServiceImpl.kt [hunk 3]
```diff
@@ -57,6 +60,10 @@ internal class OrderServiceImpl(
             }
             .flatMap { orderRepository.create(createOrderRequest.toEntity()) }
             .flatMap { order -> getCarPrice(createOrderRequest.carId).map { order.toResponse(it) } }
+            .doOnNext {
+                orderCreateOrderKafkaProducer.sendCreateRepairing(it.toProto())
```
Comment (ajax-marchenko-s) [inline domainservice/src/main/kotlin/com/makarytskyi/rentcar/service/impl/OrderServiceImpl.kt:65]: maybe log error from `orderCreateOrderKafkaProducer`

File: gateway/src/testFixtures/kotlin/com/makarytskyi/gateway/fixtures/OrderProtoFixture.kt [hunk 4]
```diff
@@ -175,8 +200,73 @@ object OrderProtoFixture {
...
-        from = Date(order.from.seconds),
-        to = Date(order.to.seconds),
+        from = timestampToDate(order.from),
+        to = timestampToDate(order.to),
         price = order.price.toBigDecimal()
     )
+
+    fun successfulGrpcGetByIdResponse(order: AggregatedOrder): GrpcGetByIdOrderResponse = GrpcGetByIdOrderResponse
+        .newBuilder()
+        .apply { successBuilder.setOrder(order).build() }
```
Comment (ajax-marchenko-s) [inline gateway/src/testFixtures/kotlin/com/makarytskyi/gateway/fixtures/OrderProtoFixture.kt]:
```suggestion
        .apply { successBuilder.order = order }
```

File: gateway/src/testFixtures/kotlin/com/makarytskyi/gateway/fixtures/OrderProtoFixture.kt [hunk 5]
```diff
@@ -175,8 +200,73 @@ object OrderProtoFixture {
...
+                order.userId = ObjectId().toString()
+            }
+        }.build()
+
+    fun successfulGrpcCreateResponse(order: Order): GrpcCreateOrderResponse = GrpcCreateOrderResponse
+        .newBuilder()
+        .apply { successBuilder.setOrder(order).build() }
+        .build()
+
+    fun failureGetFullByIdRandomResponse(exception: Exception): GetByIdOrderResponse = GetByIdOrderResponse
```
Comment (ajax-marchenko-s) [inline gateway/src/testFixtures/kotlin/com/makarytskyi/gateway/fixtures/OrderProtoFixture.kt]: this function returns `GetByIdOrderResponse`, but the name is `...GetFullById...`

File: gateway/src/testFixtures/kotlin/com/makarytskyi/gateway/fixtures/OrderProtoFixture.kt [hunk 6]
```diff
@@ -175,8 +200,73 @@ object OrderProtoFixture {
...
+        .newBuilder().also {
+            it.failureBuilder.also {
+                it.message = exception.message
+                it.notFoundBuilder
+            }
+        }
```
Comment (ajax-marchenko-s) [inline gateway/src/testFixtures/kotlin/com/makarytskyi/gateway/fixtures/OrderProtoFixture.kt]:
```suggestion
        .newBuilder().apply {
                failureBuilder.message = exception.message
                failureBuilder.notFoundBuilder
        }
```

File: internal-api/src/main/kotlin/com/makarytskyi/internalapi/topic/KafkaTopic.kt [hunk 7]
```diff
@@ -2,8 +2,10 @@ package com.makarytskyi.internalapi.topic
 
 object KafkaTopic {
     private const val REPAIRING_BASE = "repairing"
+    private const val ORDER_BASE = "order"
     private const val USER_BASE = "user"
 
+    const val ORDER_CREATE = "$ORDER_BASE.create"
```
Comment (ajax-marchenko-s) [inline internal-api/src/main/kotlin/com/makarytskyi/internalapi/topic/KafkaTopic.kt]: maybe add new object inside to group topic due to domain
so then you could write `KafkaTopic.Order.CREATE`

**Review by ajax-marchenko-s — COMMENTED** [review]
Great work!

**Review by ajax-samondros-b — APPROVED** [review]

**Pull request merged**